### PR TITLE
Feature/dsg 779/record document accepted and rejected audit events

### PR DESCRIPTION
**Pull request by jgnuvalence**
## Types of changes

Patch
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation change (documentation only, will not require redeployment)
- [ ] Refactor or cleanup (functionality unchanged)
Minor
- [ ] New feature (non-breaking change which adds functionality)
Major
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have performed a self-review of my own code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

- [ ] I have confirmed all acceptance criteria defined by the story have been met

### CODE REVIEW

File: service/src/main/java/io/nuvalence/workmanager/service/audit/transaction/AssignedToChangedAuditHandler.java [hunk 1]
```diff
@@ -47,19 +47,19 @@ public void publishAuditEvent(OffsetDateTime eventTime, String originatorId) {
             if (before == null && after != null) {
                 eventSummary =
                         String.format(
-                                "At %s, user [%s] was assigned transaction %s",
-                                eventTime, after, transactionExternalId);
+                                "User [%s] was assigned transaction %s",
```
Comment (jgnuvalence) [inline service/src/main/java/io/nuvalence/workmanager/service/audit/transaction/AssignedToChangedAuditHandler.java:50]: remove the timestamp from the summary as agreed with @JayStGelais

File: service/src/main/java/io/nuvalence/workmanager/service/audit/transaction/StatusChangedAuditHandler.java [hunk 2]
```diff
@@ -49,16 +47,16 @@ public void publishAuditEvent(OffsetDateTime eventTime, String originatorId) {
             if (before != null && !before.equals(after)) {
                 eventSummary =
                         String.format(
-                                "At %s, transaction %s changed its status to [%s]. Previously it"
+                                "Transaction %s changed its status to [%s]. Previously it"
```
Comment (jgnuvalence) [inline service/src/main/java/io/nuvalence/workmanager/service/audit/transaction/StatusChangedAuditHandler.java:50]: remove the timestamp from the summary as agreed with @JayStGelais

File: service/src/main/java/io/nuvalence/workmanager/service/domain/customerprovideddocument/CustomerProvidedDocument.java
Comment (jgnuvalence) [inline service/src/main/java/io/nuvalence/workmanager/service/domain/customerprovideddocument/CustomerProvidedDocument.java:1]: The `UpdateTrackedEntity` interface is implemented, which is a requirement to use the `AuditableAction framework,` as an implementation the `created_by`, `last_updated_by`, `created_timestamp` and `last_updated_timestamp` columns are added

File: service/src/main/java/io/nuvalence/workmanager/service/service/TransactionAuditEventService.java [hunk 3]
```diff
@@ -72,6 +75,46 @@ public AuditEventId postStateChangeEvent(
...
+     * @param timestamp time of event.
+     * @param businessObjectId id of the business object involved in event.
+     * @param businessObjectType type of the business object involved in event.
+     * @param oldStateMap map state of the business object previous to event.
+     * @param newStateMap map state of the business object after the event.
+     * @return Result audit event id.
+     * @throws ApiException for possible errors reaching audit service.
+     * @throws JsonProcessingException for possible errors converting Map states to String
+     */
+    public AuditEventId postStateChangeEvent(
```
Comment (jgnuvalence) [inline service/src/main/java/io/nuvalence/workmanager/service/service/TransactionAuditEventService.java:93]: create an overload method for postStateChangeEvent to support the cases in which we want to send multiple parameters in the pre and post states as a stringified json object

File: service/src/test/java/io/nuvalence/workmanager/service/audit/StatusChangedAuditHandlerTest.java [hunk 4]
```diff
@@ -50,7 +48,7 @@ public void testPublishAuditEvent_statusChanged() throws ApiException {
         verify(transactionAuditEventService)
                 .postStateChangeEvent(
                         eq(originatorId),
-                        eq(subjectUserId),
+                        eq(originatorId),
```
Comment (jgnuvalence) [inline service/src/test/java/io/nuvalence/workmanager/service/audit/StatusChangedAuditHandlerTest.java:51]: the id of the logged in user is sent as event userId and originatorId

File: swagger.yaml [hunk 5]
```diff
@@ -1357,8 +1357,6 @@ components:
         rejectionReason:
           type: string
           pattern: ^(POOR_QUALITY|INCORRECT_TYPE|DOES_NOT_SATISFY_REQUIREMENTS|SUSPECTED_FRAUD)$
-        dataPath:
-          type: string
```
Comment (jgnuvalence) [inline swagger.yaml:1361]: the dataPath is removed from the payload of the update customer provided documents endpoint as the queue is not needed as it is not an upsert endpoint

File: service/src/main/java/io/nuvalence/workmanager/service/service/TransactionService.java
Comment (jgnuvalence) [inline service/src/main/java/io/nuvalence/workmanager/service/service/TransactionService.java:1]: a refactor was made on this class to have the pre and post states in the controller to be used in the AuditableAction framework

**Review by asultan001 — APPROVED** [review]
Great job @jgnuvalence this LGTM

**Pull request merged**